lib/ansible/modules/cloud/amazon/kinesis_stream.py

```diff
@@ -804,6 +858,36 @@ def update(client, current_stream, stream_name, retention_period=None,
             )
             return success, changed, err_msg
 
+    if current_stream['OpenShardsCount'] != number_of_shards:
+        success, err_msg = (
+            update_shard_count(client, stream_name, number_of_shards, check_mode=check_mode)
+        )
+
+        if not success:
```

Comment:
The tag will not get updated. However, the code block above that deals with retention has the same behaviour where if the retention changes failed, it will not update the tags.

Is it Ansible module's semantics to say if a task fails, some part of the task should still finish?